client/src/modules/list/model/reducer.js

```diff
@@ -279,33 +293,124 @@ const lists = (state = {}, action) => {
         '_id'
       );
     }
-    case ItemActionTypes.ADD_SUCCESS:
-    case ItemActionTypes.ARCHIVE_SUCCESS:
+    case ListActionTypes.UPDATE_LIMIT: {
+      const {
+        payload: { listId, limit }
+      } = action;
+
+      const list = state[listId];
```

Comment:
I can see this is repeated many times in this file.

Please remove and use it straight in return statement

```
return {
        ...state,
        [listId]: {
          ...state[listId],
          listState: updatedState
        }
      };
```